From 366baf7f9d289a037ade7ab03d36acf0d4c3683d Mon Sep 17 00:00:00 2001
From: OPPEN321 <zj18139624826@gmail.com>
Date: Tue, 4 Nov 2025 09:27:17 +0800
Subject: [PATCH] The kernel has been updated to support version 6.12.

Signed-off-by: OPPEN321 <zj18139624826@gmail.com>
---
 config/Config-kernel.in | 182 +++++++++++++++++++++-------------------
 1 file changed, 95 insertions(+), 87 deletions(-)

diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index 64c8c63466..86a7804f96 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -53,7 +53,7 @@ config KERNEL_ARM_PMU
 config KERNEL_ARM_PMUV3
 	bool
 	default y if TARGET_armsr_armv8
-	depends on (arm_v7 || aarch64) && LINUX_6_6
+	depends on (arm_v7 || aarch64)
 
 config KERNEL_RISCV_PMU
 	bool
@@ -84,7 +84,7 @@ config KERNEL_X86_VSYSCALL_EMULATION
 config KERNEL_PERF_EVENTS
 	bool "Compile the kernel with performance events and counters"
 	select KERNEL_ARM_PMU if (arm || aarch64)
-	select KERNEL_ARM_PMUV3 if (arm_v7 || aarch64) && LINUX_6_6
+	select KERNEL_ARM_PMUV3 if (arm_v7 || aarch64)
 	select KERNEL_RISCV_PMU if riscv64
 
 config KERNEL_PROFILING
@@ -137,13 +137,6 @@ config KERNEL_UBSAN_BOUNDS
 	  {str,mem}*cpy() family of functions (that is addressed by
 	  FORTIFY_SOURCE).
 
-config KERNEL_UBSAN_NULL
-	bool "Enable checking of null pointers"
-	depends on KERNEL_UBSAN
-	help
-	  This option enables detection of memory accesses via a
-	  null pointer.
-
 config KERNEL_UBSAN_TRAP
 	bool "On Sanitizer warnings, abort the running kernel code"
 	depends on KERNEL_UBSAN
@@ -498,16 +491,10 @@ config KERNEL_FRAME_WARN
 # KConfig wont evaluate them unless KERNEL_EARLY_PRINTK is selected
 # which means that buildroot wont override the DEBUG_LL symbols in target
 # kernel configurations and lead to devices that dont have working console
-config KERNEL_DEBUG_LL_UART_NONE
-	bool
-	default n
-	depends on arm
-
 config KERNEL_DEBUG_LL
 	bool
 	default n
 	depends on arm
-	select KERNEL_DEBUG_LL_UART_NONE
 	help
 	  ARM low level debugging.
 
@@ -567,7 +554,7 @@ config KERNEL_BPF_EVENTS
 
 config KERNEL_PROBE_EVENTS_BTF_ARGS
 	bool "Support BTF function arguments for probe events"
-	depends on KERNEL_DEBUG_INFO_BTF && KERNEL_KPROBE_EVENTS && LINUX_6_6
+	depends on KERNEL_DEBUG_INFO_BTF && KERNEL_KPROBE_EVENTS
 
 config KERNEL_BPF_KPROBE_OVERRIDE
 	bool
@@ -607,8 +594,19 @@ choice
 
 	config KERNEL_TRANSPARENT_HUGEPAGE_MADVISE
 		bool "madvise"
+
+	config KERNEL_TRANSPARENT_HUGEPAGE_NEVER
+		bool "never"
+		depends on !LINUX_6_6
 endchoice
 
+config KERNEL_ARM64_CONTPTE
+	bool "Compile the kernel with Contiguous PTE mappings for user memory"
+	depends on KERNEL_ARM64
+	depends on KERNEL_TRANSPARENT_HUGEPAGE
+	depends on !LINUX_6_6
+	default y
+
 config KERNEL_HUGETLBFS
 	bool
 
@@ -732,10 +730,10 @@ config KERNEL_SLUB_DEBUG_ON
 	  This enables by default sanity_checks, red_zone, poison and store_user
 	  debugging options for all caches.
 
-config KERNEL_SLABINFO
-	select KERNEL_SLUB_DEBUG
-	select KERNEL_SLUB_DEBUG_ON
-	bool "Enable /proc slab debug info"
+config KERNEL_STACKDEPOT_MAX_FRAMES
+	int
+	default 64
+	depends on KERNEL_SLUB_DEBUG
 
 config KERNEL_PROC_PAGE_MONITOR
 	bool "Enable /proc page monitoring"
@@ -861,6 +859,18 @@ if KERNEL_CGROUPS
 		  Memory Nodes and assigning tasks to run only within those sets.
 		  This is primarily useful on large SMP or NUMA systems.
 
+	config KERNEL_CPUSETS_V1
+		bool "Legacy cgroup v1 cpusets controller"
+		depends on KERNEL_CPUSETS
+		depends on !LINUX_6_6
+		default n
+		help
+		  Legacy cgroup v1 cpusets controller which has been deprecated by
+		  cgroup v2 implementation. The v1 is there for legacy applications
+		  which haven't migrated to the new cgroup v2 interface yet. If you
+		  do not have any such application then you are completely fine leaving
+		  this option disabled.
+
 	config KERNEL_PROC_PID_CPUSET
 		bool "Include legacy /proc/<pid>/cpuset file"
 		depends on KERNEL_CPUSETS
@@ -872,22 +882,10 @@ if KERNEL_CGROUPS
 		  Provides a simple Resource Controller for monitoring the
 		  total CPU consumed by the tasks in a cgroup.
 
-	config KERNEL_RESOURCE_COUNTERS
-		bool "Resource counters"
-		default y
-		help
-		  This option enables controller independent resource accounting
-		  infrastructure that works with cgroups.
-
-	config KERNEL_MM_OWNER
-		bool
-		default y if KERNEL_MEMCG
-
 	config KERNEL_MEMCG
 		bool "Memory Resource Controller for Control Groups"
 		default y
 		select KERNEL_FREEZER
-		depends on KERNEL_RESOURCE_COUNTERS
 		help
 		  Provides a memory resource controller that manages both anonymous
 		  memory and page cache. (See Documentation/cgroups/memory.txt)
@@ -907,40 +905,6 @@ if KERNEL_CGROUPS
 		  This config option also selects MM_OWNER config option, which
 		  could in turn add some fork/exit overhead.
 
-	config KERNEL_MEMCG_SWAP
-		bool "Memory Resource Controller Swap Extension"
-		default y
-		depends on KERNEL_MEMCG
-		help
-		  Add swap management feature to memory resource controller. When you
-		  enable this, you can limit mem+swap usage per cgroup. In other words,
-		  when you disable this, memory resource controller has no cares to
-		  usage of swap...a process can exhaust all of the swap. This extension
-		  is useful when you want to avoid exhaustion swap but this itself
-		  adds more overheads and consumes memory for remembering information.
-		  Especially if you use 32bit system or small memory system, please
-		  be careful about enabling this. When memory resource controller
-		  is disabled by boot option, this will be automatically disabled and
-		  there will be no overhead from this. Even when you set this config=y,
-		  if boot option "swapaccount=0" is set, swap will not be accounted.
-		  Now, memory usage of swap_cgroup is 2 bytes per entry. If swap page
-		  size is 4096bytes, 512k per 1Gbytes of swap.
-
-	config KERNEL_MEMCG_SWAP_ENABLED
-		bool "Memory Resource Controller Swap Extension enabled by default"
-		depends on KERNEL_MEMCG_SWAP
-		help
-		  Memory Resource Controller Swap Extension comes with its price in
-		  a bigger memory consumption. General purpose distribution kernels
-		  which want to enable the feature but keep it disabled by default
-		  and let the user enable it by swapaccount boot command line
-		  parameter should have this option unselected.
-
-		  Those who want to have the feature enabled by default should
-		  select this option (if, for some reason, they need to disable it,
-		  then swapaccount=0 does the trick).
-
-
 	config KERNEL_MEMCG_KMEM
 		bool "Memory Resource Controller Kernel Memory accounting (EXPERIMENTAL)"
 		default y
@@ -952,6 +916,22 @@ if KERNEL_CGROUPS
 		  Memory Controller, which are page-based, and can be swapped. Users of
 		  the kmem extension can use it to guarantee that no group of processes
 		  will ever exhaust kernel resources alone.
+	
+	config KERNEL_MEMCG_V1
+		bool "Legacy cgroup v1 memory controller"
+		default n
+		depends on KERNEL_MEMCG
+		depends on !LINUX_6_6
+		help
+		  Legacy cgroup v1 memory controller which has been deprecated by
+		  cgroup v2 implementation. The v1 is there for legacy applications
+		  which haven't migrated to the new cgroup v2 interface yet. If you
+		  do not have any such application then you are completely fine leaving
+		  this option disabled.
+
+		  Please note that feature set of the legacy memory controller is likely
+		  going to shrink due to deprecation process. New deployments with v1
+		  controller are highly discouraged.
 
 	config KERNEL_CGROUP_PERF
 		bool "Enable perf_event per-cpu per-container group (cgroup) monitoring"
@@ -1096,16 +1076,6 @@ if KERNEL_NAMESPACES
 
 endif
 
-config KERNEL_DEVPTS_MULTIPLE_INSTANCES
-	bool "Support multiple instances of devpts"
-	default y if !SMALL_FLASH
-	help
-	  Enable support for multiple instances of devpts filesystem.
-	  If you want to have isolated PTY namespaces (eg: in containers),
-	  say Y here. Otherwise, say N. If enabled, each mount of devpts
-	  filesystem with the '-o newinstance' option will create an
-	  independent PTY namespace.
-
 config KERNEL_POSIX_MQUEUE
 	bool "POSIX Message Queues"
 	default y if !SMALL_FLASH
@@ -1212,6 +1182,28 @@ config KERNEL_NET_L3_MASTER_DEV
 	  drivers to support L3 master devices like VRF.
   	  Increases the compressed kernel size by ~4kB (as of Linux 6.6).
 
+config KERNEL_DCB
+	bool "Data Center Bridging support"
+	default y if TARGET_armsr_armv8
+	default y if TARGET_x86_64
+	help
+	  This enables support for configuring Data Center Bridging (DCB)
+	  features on DCB capable Ethernet adapters via rtnetlink.  Say 'Y'
+	  if you have a DCB capable Ethernet adapter which supports this
+	  interface and you are connected to a DCB capable switch.
+
+	  DCB is a collection of Ethernet enhancements which allow DCB capable
+	  NICs and switches to support network traffic with differing
+	  requirements (highly reliable, no drops vs. best effort vs. low
+	  latency) to co-exist on Ethernet.
+
+	  DCB features include:
+	    Enhanced Transmission Selection (aka Priority Grouping) - provides a
+	      framework for assigning bandwidth guarantees to traffic classes.
+	    Priority-based Flow Control (PFC) - a MAC control pause frame which
+	      works at the granularity of the 802.1p priority instead of the
+	      link (802.3x).
+
 config KERNEL_XDP_SOCKETS
 	bool "XDP sockets support"
 	help
@@ -1286,6 +1278,23 @@ config KERNEL_BTRFS_FS
 	  Say Y here if you want to make the kernel to be able to boot off a
 	  BTRFS partition.
 
+config KERNEL_EROFS_FS
+	bool "Compile the kernel with built-in EROFS support"
+	help
+	  Say Y here if you want to make the kernel to be able to boot off a
+	  EROFS partition.
+
+config KERNEL_EROFS_FS_XATTR
+	bool "EROFS XATTR support"
+
+config KERNEL_EROFS_FS_ZIP
+	bool
+	default y if KERNEL_EROFS_FS
+
+config KERNEL_EROFS_FS_ZIP_LZMA
+	bool
+	default y if KERNEL_EROFS_FS
+
 menu "Filesystem ACL and attr support options"
 	config USE_FS_ACL_ATTR
 		bool "Use filesystem ACL and attr support by default"
@@ -1306,6 +1315,11 @@ menu "Filesystem ACL and attr support options"
 		select KERNEL_FS_POSIX_ACL
 		default y if USE_FS_ACL_ATTR
 
+	config KERNEL_EROFS_FS_POSIX_ACL
+		bool "Enable POSIX ACL for EROFS Filesystems"
+		select KERNEL_FS_POSIX_ACL
+		default y if USE_FS_ACL_ATTR
+
 	config KERNEL_EXT4_FS_POSIX_ACL
 		bool "Enable POSIX ACL for Ext4 Filesystems"
 		select KERNEL_FS_POSIX_ACL
@@ -1378,13 +1392,6 @@ config KERNEL_DEVMEM
 	  The /dev/mem device is used to access areas of physical
 	  memory.
 
-config KERNEL_DEVKMEM
-	bool "/dev/kmem virtual device support"
-	help
-	  Say Y here if you want to support the /dev/kmem device. The
-	  /dev/kmem device is rarely used, but can be used for certain
-	  kind of kernel debugging operations.
-
 config KERNEL_SQUASHFS_FRAGMENT_CACHE_SIZE
 	int "Number of squashfs fragments cached"
 	default 2 if (SMALL_FLASH && !LOW_MEMORY_FOOTPRINT)
@@ -1435,10 +1442,6 @@ config KERNEL_SECURITY_SELINUX_BOOTPARAM
 	depends on KERNEL_SECURITY_SELINUX
 	default y
 
-config KERNEL_SECURITY_SELINUX_DISABLE
-	bool "NSA SELinux runtime disable"
-	depends on KERNEL_SECURITY_SELINUX
-
 config KERNEL_SECURITY_SELINUX_DEVELOP
 	bool "NSA SELinux Development Support"
 	depends on KERNEL_SECURITY_SELINUX
@@ -1459,6 +1462,11 @@ config KERNEL_LSM
 	default "lockdown,yama,loadpin,safesetid,integrity,selinux"
 	depends on KERNEL_SECURITY_SELINUX
 
+config KERNEL_EROFS_FS_SECURITY
+	bool "EROFS Security Labels"
+	default y if !SMALL_FLASH
+	select KERNEL_EROFS_FS_XATTR
+
 config KERNEL_EXT4_FS_SECURITY
 	bool "Ext4 Security Labels"
 	default y if !SMALL_FLASH
-- 
2.43.0
